packages/util/test/internal.spec.ts

```diff
@@ -67,4 +69,62 @@ describe('internal', () => {
     assert.equal(isHexString('0x0000000000000000000000000000000000000000'), true)
     assert.equal(isHexString('123'), false)
   })
+
+  describe('isHexPrefixed', () => {
+    it('should return true for hex-prefixed string', () => {
+      assert.isTrue(isHexPrefixed('0x123'))
+    })
+
+    it('should return false for non-hex-prefixed string', () => {
+      assert.isFalse(isHexPrefixed('123'))
+    })
+  })
+
+  describe('padToEven', () => {
+    it('should pad odd-length string to even', () => {
+      assert.equal(padToEven('123'), '0123')
+    })
+
+    it('should not pad even-length string', () => {
+      assert.equal(padToEven('1234'), '1234')
+    })
+  })
+
+  describe('getBinarySize', () => {
+    it('should return the correct binary size of a string', () => {
+      assert.equal(getBinarySize('Hello, World!'), 13)
+    })
+  })
+
+  describe('arrayContainsArray', () => {
+    it('should return true when the first array contains all elements of the second', () => {
+      assert.isTrue(arrayContainsArray([1, 2, 3, 4, 5], [3, 4]))
+    })
+
+    it('should return false when the first array does not contain all elements of the second', () => {
+      assert.isFalse(arrayContainsArray([1, 2, 3, 4, 5], [6, 7]))
+    })
+  })
+
+  describe('fromUtf8', () => {
+    it('should convert a UTF-8 string to a hex string', () => {
+      assert.equal(fromUtf8('Hello, World!'), '0x48656c6c6f2c20576f726c6421')
```

Comment:
This string only contains ASCII characters. Can we also test an UTF8 string here? Ref: https://www.fileformat.info/info/charset/UTF-8/list.htm 

(So this visually adds one character, but actually adds two bytes in the string ref)

Reply:
Created a new test that includes UTF8 2-byte characters.